File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    gameboard = []
    for i in moves:
        for j in range(len(board)):
            if board[j][i-1] == 0:
                pass
            else:
                gameboard.append(board[j][i-1])
                board[j][i-1] = 0
                
                if len(gameboard) > 1:
                    if gameboard[-1] == gameboard[-2]:
                        gameboard.pop(-1)
                        gameboard.pop(-1)
                        answer +=2
                break
    
        
    print(gameboard)
    print(board)
    print(answer)
    return answer

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Stack;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        for(int i : moves){
            for(int j=0; j<board.length; j++){
                if(board[j][i-1] == 0){
                    continue;
                }    
                else{
                    if(stack.peek() == board[j][i-1]){
                        
                        answer +=2;
                        stack.pop();
                        
                    }
                    else{
                        stack.push(board[j][i-1]);
                    }
                    board[j][i-1] = 0;
                    break;
                    
                    
                }
            }
        }
        return answer;
    }
}